incubator/hnc/pkg/reconcilers/hnc_config_test.go

```diff
@@ -120,36 +118,18 @@ func hasTypeWithMode(apiVersion, kind string, mode api.SynchronizationMode, conf
 }
 
 func makeSecret(ctx context.Context, nsName, secretName string) {
-	sec := &corev1.Secret{}
-	sec.Name = secretName
-	sec.Namespace = nsName
-	ExpectWithOffset(1, k8sClient.Create(ctx, sec)).Should(Succeed())
+	makeObject(1, ctx, GVKs["Secret"], nsName, secretName)
```

Comment:
This is a great cleanup! But I wonder if it could be even better - get rid of all the type-specific functions, and also allow you to pass in the string directly instead of using the `GVKs` map? Eg, instead of calling `makeSecret(ctx, nnm, nm)`, just call `makeObject(ctx, "Secret", nnm, nm)`? That will make this change a bit messier as you modify all the call sites, but much better after that.

Reply:
Good idea. I agree it is better to get rid of them once and for all. Please see the updated code.